spec/models/weight_limits_spec.rb

```diff
@@ -91,6 +101,171 @@ def build_inventory_unit(variant, order)
       Spree::ActiveShipping::Config.set(:default_weight => 1)
     end
 
+    describe ".valid_weight_for_package?" do
+      context "with a non nil max weight" do
+        it "should return true if the max_weight is equal to zero" do
+          expect(domestic_calculator.send :valid_weight_for_package?, package, 0).to eq true
+        end
+
+        it "should return true if the package weight is equal to the supplied max_weight" do
+          allow(package).to receive(:weight).and_return(1)
+          expect(domestic_calculator.send :valid_weight_for_package?, package, 1).to eq true
+        end
+
+        it "should return true if the package weight is less than the supplied max_weight" do
+          allow(package).to receive(:weight).and_return(1)
+          expect(domestic_calculator.send :valid_weight_for_package?, package, 2).to eq true
+        end
+
+        it "should return false if the package weight is greater than the supplied max_weight" do
+          allow(package).to receive(:weight).and_return(3)
+          expect(domestic_calculator.send :valid_weight_for_package?, package, 2).to eq false
+        end
+      end
+
+      context "with a nil max weight" do
+        it "should return false if the max_weight is nil" do
+          expect(domestic_calculator.send :valid_weight_for_package?, package, nil).to eq false
+        end
+      end
+    end
+
+    describe ".country_weight_error?" do
+      context "with a package that is too heavy" do
+        it "should raise a Spree::ShippingError" do
+          allow(domestic_calculator).to receive(:valid_weight_for_package?).and_return(false)
+          expect{ domestic_calculator.send :country_weight_error?, package }.to raise_error(Spree::ShippingError)
+        end
+      end
+
+      context "with a package within the weights limit" do
+         it "should return false" do
+          allow(domestic_calculator).to receive(:valid_weight_for_package?).and_return(true)
+          expect(domestic_calculator.send :country_weight_error?, package).to be_nil
+        end
+      end
+    end
+
+    describe ".convert_package_to_weights_array" do
```

Comment:
This should be tested as part of the tests for `ActiveShipping::Base#compute_package`. Having said that testing all this as part of `#compute_package` will probably be quite messy, so I would be in favor of extract the package weights calculation to a separate class, as it isn't the calculators job to deal with all this. I know that the goal is to have tests for everything before we start refactoring, but in this particular case I think extracting the behavior and adding the tests in the same step may be easier.